pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/SpoolIntegrationTest.java

```diff
@@ -117,13 +121,146 @@ public void intermediateSpool()
     JsonNode stats = jsonNode.get("stageStats");
     assertNoError(jsonNode);
     DocumentContext parsed = JsonPath.parse(stats.toString());
-    List<Map<String, Object>> stage4On3 = parsed.read("$..[?(@.stage == 3)]..[?(@.stage == 4)]");
-    Assert.assertEquals(stage4On3.size(), 1, "Stage 4 should be descended from stage 3 exactly once");
 
-    List<Map<String, Object>> stage4On7 = parsed.read("$..[?(@.stage == 7)]..[?(@.stage == 4)]");
-    Assert.assertEquals(stage4On3.size(), 1, "Stage 4 should be descended from stage 7 exactly once");
+    checkSpoolTimes(parsed, 4, 3, 1);
+    checkSpoolTimes(parsed, 4, 7, 1);
+    checkSpoolSame(parsed, 4, 3, 7);
+  }
+
+  /**
+   * Test a complex with nested spools. Don't try to understand it, just check that the spools are correct.
+   * The test name corresponds to the PR that fixed the issue.
+   */
+  @Test
+  public void test15135()
```

Comment:
I'm not sure why the PR number is relevant here - can't we just call this something like `testMultiSpools`?

Reply:
In general, it is a good practice to include the number where the issue was detected, but in this case, I'm using this because... I have no clue what this query is testing. The only thing I know is that it was the simplified version of a query that detected the errors fixed here.

I don't know if `testMultiSpools` is correct. More tests with multi-spools can be created in the future. Anyway, I can change it if you think it is useful.

Reply:
changed